packages/contracts-bedrock/contracts/L2/L2CrossDomainMessenger.sol

```diff
@@ -49,14 +50,26 @@ contract L2CrossDomainMessenger is CrossDomainMessenger, Semver {
      * @inheritdoc CrossDomainMessenger
      */
     function _sendMessage(
+        uint32 _type,
         address _to,
         uint64 _gasLimit,
         uint256 _value,
         bytes memory _data
     ) internal override {
-        L2ToL1MessagePasser(payable(Predeploys.L2_TO_L1_MESSAGE_PASSER)).initiateWithdrawal{
-            value: _value
-        }(_to, _gasLimit, _data);
+        if (_type == BridgeConstants.ETH_WITHDRAWAL_TX){
+            L2ToL1MessagePasser(payable(Predeploys.L2_TO_L1_MESSAGE_PASSER)).initiateWithdrawal{
+                    value: 0
+                }(_type,_value,_to, _gasLimit, _data);
+        }else if (_type == BridgeConstants.MNT_WITHDRAWAL_TX){
+            L2ToL1MessagePasser(payable(Predeploys.L2_TO_L1_MESSAGE_PASSER)).initiateWithdrawal{
+                    value: _value
+                }(_type,_value,_to, _gasLimit, _data);
+        }else{
+            L2ToL1MessagePasser(payable(Predeploys.L2_TO_L1_MESSAGE_PASSER)).initiateWithdrawal{
+                    value: _value
+                }(_type,_value,_to, _gasLimit, _data);
+        }
+
```

Comment:
The external ETH withdraw is not payable. So the value must be zero. For other case, you just specify the same code. So I think there is no necessary to just `_type` here.